Automate deletion of temporary files on windows

aj062 opened the pull request:
#### a7ec0ec651a5c6c12cdd1aeee9b05bde45557fc9
<pre>
Automate deletion of temporary files on windows
<a href="https://bugs.webkit.org/show_bug.cgi?id=245786">https://bugs.webkit.org/show_bug.cgi?id=245786</a>
rdar://72849252

Reviewed by Ryan Haddad.

* Tools/CISupport/delete-stale-build-files:
* Tools/CISupport/ews-build/factories.py:
* Tools/CISupport/ews-build/steps.py:
(DeleteStaleBuildFiles):
* Tools/CISupport/ews-build/steps_unittest.py: Added unit-tests.
* Tools/CISupport/ews-build/factories_unittest.py: Updated unit-tests.

Canonical link: <a href="https://commits.webkit.org/255045@main">https://commits.webkit.org/255045@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1c584cfa5b2518ccbf6cbb4e9d57d2a3f1702a03

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/90882 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/35453 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/21432 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/100165 "Built successfully") | [❌ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/158719 "Reverted pull request changes (failure)") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/94888 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/33956 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/29009 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/83224 "Built successfully") | [💥 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/97050 "An unexpected error occured. Recent messages:Failed to print configuration") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/96538 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/27055 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/77677 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/26864 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/81808 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/81544 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/69895 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/35037 "Built successfully") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/15574 "Found 1 new test failure: webanimations/accelerated-web-animation-with-single-interval-and-easing-y-axis-above-1.html (failure)") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/89948 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/32838 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/16557 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/36615 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/39508 "Found 1 new test failure: fast/events/blur-remove-parent-crash.html (failure)") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1545 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/38541 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/35646 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

chgibb-apple commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -5254,3 +5254,16 @@ def doStepIf(self, step):
+    name = 'delete-stale-build-files'
+    description = ['Deleting stale build files']
+    descriptionDone = ['Deleted stale build files']
+    command = ['python3', 'Tools/CISupport/delete-stale-build-files', WithProperties('--platform=%(fullPlatform)s')]
```
`delete-stale-build-files` looks like it's expecting `--platform` to be `win`. Does `fullPlatform` report `win` on Windows or a "fuller" double like `win32-x64`?

aj062 replied:
Yes, fullPlatform is "win". It can be verified in any ews build like https://ews-build.webkit.org/#/builders/10/builds/158586 (in properties tab)

roynos commented on Tools/CISupport/delete-stale-build-files:
```diff
@@ -86,6 +91,21 @@ def webkitBuildDirectory(platform, fullPlatform, configuration):
         "--" + platform, "--" + configuration, '--top-level'], stdout=subprocess.PIPE).communicate()[0].strip()
 
+def deleteWindowsStaleFiles():
+    directoriesToDelete = ['/tmp',
```
Blindly deleting `/tmp` is not a good idea.

chgibb-apple replied:
Does `/tmp` exist on Windows? Would temp not be something like `/cygdrive/c/buildbot/AppData/local/Temp`? Given there is no "root" (rather only drive mount points) on NTFS.

roynos commented on Tools/CISupport/delete-stale-build-files:
```diff
@@ -86,6 +91,21 @@ def webkitBuildDirectory(platform, fullPlatform, configuration):
+def deleteWindowsStaleFiles():
+    directoriesToDelete = ['/tmp',
+                           '/cygdrive/c/Program Files (x86)/Windows Kits/10/Debuggers/x64/sym/WebKit.pdb',
+                           '/cygdrive/c/Program Files (x86)/Windows Kits/10/Debuggers/x64/sym/WTF.pdb',
```
I think it's possible to determine the `Debuggers` folder with environment variable. It will make this good resilient to updates.

JonWBedard replied:
Any reason not to just delete all the Debuggers?

aj062 replied:
No reason, I wasn't sure if it's fine to delete it totally (e.g.: not sure if those files are used as a cache to speed up subsequent builds). Those were the directories which were large (in GBs, rest all were smaller). e.g.:

buildbot@ews210 /cygdrive/c/Program Files (x86)/Windows Kits/10/Debuggers/x64/sym
$ du -hs * | grep G
21G     DumpRenderTreeLib.pdb
21G     JavaScriptCore.pdb
4.1G    WTF.pdb
133G    WebKit.pdb

ryanhaddad reviewed (approved):
Based on a slack conversation, it may be worth investigating how we can disable the generation of unnecessary symbol files at build time, but with how frequently our Windows bots are running out of space and requiring manual intervention this seems like a good stopgap solution.

aj062 committed:
Automate deletion of temporary files on windows
https://bugs.webkit.org/show_bug.cgi?id=245786
rdar://72849252

Reviewed by Ryan Haddad.

* Tools/CISupport/delete-stale-build-files:
* Tools/CISupport/ews-build/factories.py:
* Tools/CISupport/ews-build/steps.py:
(DeleteStaleBuildFiles):
* Tools/CISupport/ews-build/steps_unittest.py: Added unit-tests.
* Tools/CISupport/ews-build/factories_unittest.py: Updated unit-tests.

Canonical link: https://commits.webkit.org/255045@main
